docs/cloud-messaging/receive.md

```diff
@@ -409,3 +409,97 @@ experimentalSetDeliveryMetricsExportedToBigQueryEnabled(messaging, true);
 ```
 
 Don't forget to run `yarn build` in order to export the new version of your service worker to the `web` folder.
+
+
+## (Advanced, Optional) Allowing Notification Images on iOS
+
+On Apple devices, in order for incoming FCM Notifications to display images from the FCM payload, you must add an additional notification service extension. This is not a required step.
```

Comment:
Since you proceed with four discrete steps, it could be good to mention each one of them in this intro.  As this is, it seems to suggest adding the extension is the only task.

Also: omit "This is not a required step" to disambiguate. I don't think readers will assume they have to do this unless they want to display images this way.

Reply:
I've added 'and configure your app to use it' to help the user understand that there are several steps